### PR TITLE
Prevent refresh on enter in header search (documents and images)

### PR DESCRIPTION
**Pull request by rd3m**
References issue #3746

* Do the tests still pass? (https://docs.wagtail.org/en/latest/contributing/developing.html#testing) - yes
* Does the code comply with the style guide? (Run `make lint` from the Wagtail root) - yes
* For front-end changes: Did you test on all of [Wagtail’s supported environments](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)?
    * **Please list the exact browser and operating system versions you tested**.
      * ios Phone and Tablet - latest Safari
      * Chrome Desktop - latest
      * Safari macOs - latest
      * Firefox Desktop - latest
    * **Please list which assistive technologies you tested**.
      * n/a

### CODE REVIEW

**Comment by lb-** [comment]
@rd3m 
Not sure that a JavaScript solution is ideal here, maybe we should explore something like this 
https://stackoverflow.com/a/51507806/8070948

**Comment by rd3m** [comment]
Hi @lb-, I explored the option of using a disabled hidden submit button but it looks like the search query is running on `'keyup cut paste change'` rather than form submission, so I thought it was better to prevent the key stroke from reaching the field. Let me know your thoughts!

**Comment by rd3m** [comment]
Hi @lb-, yes, that's a much nicer solution.

**Comment by lb-** [comment]
not for review - need to re-check keyboard and screen reader accessibility for this change

**Review by lb- — APPROVED** [review]
Looks great @rd3m - thanks for taking the time to try out some different solutions.

Keyboard control still works when focused on the hidden button (the non-disabled one) and screen reader interaction still works.

Thanks for making your first contribution to Wagtail!

I have put your name on the changelog and contributors list as (Riley de Mestre) - please let me know if you would like your name listed differently.

**Comment by lb-** [comment]
Added changelog/pushed to remote branch.

File: wagtail/admin/templates/wagtailadmin/shared/header.html [hunk 1]
```diff
@@ -29,6 +29,8 @@ <h1>{% if icon %}{% icon name=icon class_name="header-title-icon" %}{% endif %}
                         {% for field in search_form %}
                             {% include "wagtailadmin/shared/field_as_li.html" with field=field field_classes="field-small iconfield" icon="search" %}
                         {% endfor %}
+                        {% comment %}Ad an initial disabled & hidden submit button so that pressing 'enter' will not submit form. Reload not required as the content is dynamically loaded.{% endcomment %}
```
Comment (tomdyson) [inline wagtail/admin/templates/wagtailadmin/shared/header.html:32]: Ad an -> Add an

Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/shared/header.html:32]: thanks, will fix up

Comment (lb-) [reply wagtail/admin/templates/wagtailadmin/shared/header.html:32]: https://github.com/wagtail/wagtail/commit/93a1ffe5dac5a05ee8df51e9e6710b9897086058

**Comment by rd3m** [comment]
First contribution, very exciting, thanks @lb- !

**Pull request merged**